account_vat_period_end_statement/tests/test_vat_statement.py

```diff
@@ -105,6 +105,46 @@ def setUp(self):
                 "amount_type": "percent",
                 "vat_statement_account_id": self.received_vat_account,
                 "type_tax_use": "sale",
+                "invoice_repartition_line_ids": [
```

Comment:
Ho provato in locale e il test qui sotto passa anche senza la fix che hai fatto in `l10n_it_account`; se aggiungi una fix sarebbe meglio che i test falliscano senza la fix, probabilmente è sufficiente aggiungere un `assert` mirato nel test qui sotto.

Potresti verificare?

Reply:
Il punto era che, con la modifica su `l10n_it_account`, i test di `account_vat_period_end_statement` fallivano perchè le imposte venivano create in modo diverso da quanto avviene quando crei un DB manualmente.

Ora, modificando come suggerisci in https://github.com/OCA/l10n-italy/pull/2277/files#r636741552 , questa modifica non è più necessaria, quindi ho rimosso questo file.

Bisogna però tenere conto che in questo modo i test automatici sono appunto più distanti dagli scenari reali

Reply:
Capito, quindi ora abbiamo due opzioni:
1. Ripristinare la modifica ai test di `account_vat_period_end_statement` e aggiungere un `assert` opportuno per verificare la fix fatta in `l10n_it_account`
2. Rimuovere la modifica a `account_vat_period_end_statement` e lasciare i suoi test automatici
    > più distanti dagli scenari reali

    (cosa che hai già fatto)
 
Riesci ad implementare la prima opzione?
Giusto perché così abbiamo dei test più verosimili in `account_vat_period_end_statement`.

Approvo a prescindere perché la modifica aggiunta a `l10n_it_account` ha già un suo test e la modifica chiesta per `account_vat_period_end_statement` è fuori dallo scope della PR.

Reply:
Con l'implementazione attuale di `get_balance_domain` stiamo già coprendo casi distanti degli scenari reali; quindi non mi viene in mente un assert da far fallire.

Voglio dire:

 - Nei test le imposte vengono create senza `invoice_repartition_line_ids`, nel modulo `l10n_it` con
 - L'implementazione di `get_balance_domain` con `if account_ids` gestisce il caso in cui le imposte vengono create senza `invoice_repartition_line_ids`

Quali siano gli scenari reali con imposte senza `invoice_repartition_line_ids` non ne ho idea.

L'unico caso che mi viene in mente è quello delle migrazioni da versioni precedenti

Reply:
Per questo la prima opzione iniziava con:
> Ripristinare la modifica ai test di `account_vat_period_end_statement`

Quindi riaggiungendo https://github.com/OCA/l10n-italy/compare/f7c103a6e2db3b9537fd880a0bbde9519780583d..d0075eed5f09fef8882d09085a2f329445fd436a
in modo da avere nei test di `account_vat_period_end_statement` uno scenario più realistico.

L'`assert` da aggiungere nel test di `account_vat_period_end_statement` sarebbe così su uno scenario realistico

Reply:
Cioè che assert?

Reply:
Se ho capito bene lo scopo della PR, allora il risultato del metodo `report.account_vat_period_end_statement.vat_statement._get_taxes_amounts` dovrebbe fare la ripartizione tra `vat_deductible` e `vat_undeductible` in modo non corretto, quindi potresti mettere un assert che ne controlla il risultato in modo che fallisca senza le modifiche in questa PR.

Oppure, visto che nella issue indichi i valori delle righe `account.vat.period.end.statement.credit_vat_account_line_ids` come sbagliati, potresti verificare che con le modifiche in questa PR vengono invece corretti.

Reply:
Se mai ne parliamo a voce